src/int/mul.rs

```diff
@@ -31,13 +31,30 @@ impl<const LIMBS: usize> Int<LIMBS> {
 
         (lo, hi, negate)
     }
+
+    /// Multiply `self` by `rhs`, returning a concatenated "wide" result.
+    pub const fn widening_mul<const RHS_LIMBS: usize, const WIDE_LIMBS: usize>(
```

Comment:
Now that we have `widening_mul`, could be nice to implement `Mul<Int<RHS_LIMBS>>` for `Int<LIMBS>` similarly to `Uint`. This was done by me at the time using macros, the idea was for the standard type to have `Mul<Int128> for Int192` and such.